File: src/test/java/automation_projects/automation/youtube.java
package automation_projects.automation;

import org.junit.Assert;
import org.openqa.selenium.By;
import org.openqa.selenium.Keys;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.chrome.ChromeOptions;
import org.testng.annotations.AfterClass;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.BeforeMethod;
import org.testng.annotations.Test;

import java.util.concurrent.TimeUnit;

public class youtube {

    WebDriver driver;
    ChromeOptions options;
    @BeforeClass
    public void settingUp(){
        options = new ChromeOptions();
        options.addArguments("--lang=en-GB");
        options.addArguments("--disable-blink-features=AutomationControlled");
        options.addArguments("--start-maximized");
        driver = new ChromeDriver(options);
        driver.manage().timeouts().implicitlyWait(10, TimeUnit.SECONDS);

    }

    @Test
    public void automatingYoutube(){

        driver.get("https://www.youtube.com");
        driver.findElement(By.xpath("//input[@*='ytSearchboxComponentInput yt-searchbox-input title']")).sendKeys("Eminem" + Keys.ENTER);
        driver.findElement(By.xpath("//*[@*='Eminem - Lose Yourself [HD]']")).click();

        String actualText = "Eminem - Lose Yourself [HD]";
        String expectedText = driver.findElement(By.xpath("//a[contains(@title, 'Eminem - Lose Yourself [HD]')]")).getText();
        System.out.println("SONG IS NAME -----> " + expectedText);
        Assert.assertEquals("TEXT IS NOT MATCHING", expectedText, actualText);



    }


    @AfterClass
    public void shuttingDown() throws InterruptedException {
        Thread.sleep(3000);
        driver.quit();
    }
}

File: my_automation/automation/youtube.py
import time
import unittest

from selenium import webdriver
from selenium.webdriver import Keys
from selenium.webdriver.chrome.options import Options
from  selenium.webdriver.common.by import By


class MyYouTube(unittest.TestCase):

    def setUp(self):
        options = Options()
        options.add_experimental_option("detach", True)
        options.add_argument("--disable-blink-features=AutomationControlled")
        options.add_argument("enable-automation")
        options.add_argument("--start-maximized")
        options.add_argument("--lang=en-GB")
        self.driver = webdriver.Chrome(options)
        self.driver.implicitly_wait(10)

    def test_youtube_video(self):
        driver = self.driver
        driver.get("https://youtube.com")
        driver.find_element(By.XPATH, "//input[@*='ytSearchboxComponentInput yt-searchbox-input title']").send_keys("Eminem" + Keys.ENTER)
        eminemVideo = driver.find_element(by=By.XPATH, value="//*[@*='Eminem - Lose Yourself [HD]']")
        eminemVideo.click()
        expectedYouTubeText = driver.find_element(By.XPATH, "//a[contains(@title, 'Eminem - Lose Yourself [HD]')]").text
        print(expectedYouTubeText)
        assert expectedYouTubeText == "Eminem - Lose Yourself [HD]"

    def tearDown(self):
        time.sleep(3)
        self.driver.quit()


if __name__ == '__main__':
    unittest.main()
